### PR TITLE
Add ability to flush

### PR DESCRIPTION
**Pull request by withinboredom**
This one is a bit tricky to test automatically... any tips/suggestions on how to do so would be highly appreciated. Instead, I had to resort to manual tests. Here is the code I ran:

```php
<?php

ini_set('output_buffering', '0');

header('Content-Type: text/plain');

echo "immediate\n";
flush();
echo "later?\n";
sleep(2);
flush();
sleep(3);
echo "more later\n";
die();
```

Running locally with `curl localhost/test.php -v` should output something like this:

```
> GET / HTTP/2
> Host: localhost
> user-agent: curl/7.81.0
> accept: */*
>
< HTTP/2 200
< alt-svc: h3=":443"; ma=2592000
< content-type: text/plain;charset=UTF-8
< server: Caddy
< x-powered-by: PHP/8.2.0RC5
< date: Mon, 07 Nov 2022 23:23:21 GMT
<
immediate
SLEEP
SLEEP
later?
SLEEP
SLEEP
SLEEP
more later
```

Removing the flushes should result in a 5s delay before seeing any output.

NOTE: I also found a few typos and fixed them here. Let me know if it makes things too noisy, and I can move it to a different PR.

### CODE REVIEW

File: frankenphp.c [hunk 1]
```diff
@@ -448,6 +448,20 @@ static size_t frankenphp_read_post(char *buffer, size_t count_bytes)
...
+static void frankenphp_flush(void *server_context)
+{
+	frankenphp_server_context* ctx = SG(server_context);
+
+	sapi_send_headers();
+
+	if(ctx->finished) {
+		// todo: do we have a signal from go that the connection is aborted? If so:
+		// php_handle_aborted_connection();
+	}
```
Comment (withinboredom) [inline frankenphp.c:460]: I imagine the whole block will be optimized out during compilation, but we can comment the whole thing out (or remove it).

File: frankenphp.go [hunk 2]
```diff
@@ -500,6 +500,14 @@ func go_read_post(rh C.uintptr_t, cBuf *C.char, countBytes C.size_t) C.size_t {
 	return C.size_t(readBytes)
 }
 
+//export go_flush_response
+func go_flush_response(rh C.uintptr_t) {
+	r := cgo.Handle(rh).Value().(*http.Request)
+	fc := r.Context().Value(contextKey).(*FrankenPHPContext)
+	fc.responseWriter.(http.Flusher).Flush()
+	println("flushed")
```
Comment (withinboredom) [inline frankenphp.go:508]: This is here for now, for testing.

Future @withinboredom: remove this before merging.

**Comment by withinboredom** [comment]
oh yes, segfaults all over the place. I knew it couldn't be that simple :D I'll take a look in the morning.

**Comment by dunglas** [comment]
Sorry I didn't see your PR before opening mine (I was trying to reproduce your issue in the meantime): https://github.com/dunglas/frankenphp/pull/90

**Comment by withinboredom** [comment]
Aha! it's all good and yours is better! I'll close this :)

**Pull request closed without merging**